src/authenticator.ts

```diff
@@ -13,28 +8,40 @@ export interface AuthenticateCallback<User> {
 /**
  * Extra options for the authenticator.
  */
-export interface AuthenticatorOptions {
-  sessionKey?: AuthenticateOptions["sessionKey"];
-  sessionErrorKey?: AuthenticateOptions["sessionErrorKey"];
-  sessionStrategyKey?: AuthenticateOptions["sessionStrategyKey"];
-  throwOnError?: AuthenticateOptions["throwOnError"];
+export interface PublicAuthenticatorOptions<TUser, TFlash, TContext> {
+  sessionStorage: SessionStorage<TUser, TFlash>;
+  sessionKey?: AuthenticateOptions<TContext>["sessionKey"];
+  sessionErrorKey?: AuthenticateOptions<TContext>["sessionErrorKey"];
+  sessionStrategyKey?: AuthenticateOptions<TContext>["sessionStrategyKey"];
+  throwOnError?: AuthenticateOptions<TContext>["throwOnError"];
 }
```

Comment:
This adds the sessionStorage to the options to bring things down to a single required options bag. This is breaking (but then again, much of this is), but has proven to be a good change in our TanStack libraries.

Reply:
This can be a non-breaking change if the current API is still implemented by the RemixAuthenticator